ext/spl/php_spl.c

```diff
@@ -523,44 +523,45 @@ PHP_FUNCTION(spl_autoload_register)
 			if (Z_TYPE_P(zcallable) == IS_ARRAY) {
 				if (!obj_ptr && alfi.func_ptr && !(alfi.func_ptr->common.fn_flags & ZEND_ACC_STATIC)) {
 					if (do_throw) {
-						zend_throw_exception_ex(spl_ce_LogicException, 0, "Passed array specifies a non static method but no object (%s)", error);
+						zend_type_error("Passed array specifies a non static method but no object (%s)", error);
 					}
 					if (error) {
 						efree(error);
 					}
 					zend_string_release_ex(func_name, 0);
-					RETURN_FALSE;
+					return;
```

Comment:
All these `return`s should be `RETURN_THROWS()` https://github.com/php/php-src/pull/5036

Reply:
Huh, I though that you went through them that's why I needed to manually fix the merge conflicts. Will do the remaining ones

Reply:
Ah, no, please ignore. I didn't add `RETURN_THROWS()` previously myself here, because it's not sure that an exception is thrown. But then what is the purpose of changing the return type?

Reply:
I didn't look at the specific case until I opened my editor. But yeah now I'm not sure why I changed the return type ...

Reply:
Maybe
````c
if (do_throw) {
    RETURN_THROWS();
} else {
    RETURN_FALSE;
}
````

Reply:
I think I've covered all the case, please let me know if there are some that I missed.

Reply:
For me, the PR itself looks fine (apart from the few issues with `RETURN_THROWS()`), but we should try to address Nikita's concerns. :)